src/main/sensors/gyro.c

```diff
@@ -533,6 +539,41 @@ bool gyroInit(void)
     return ret;
 }
 
+#ifdef USE_DYN_LPF
+static FAST_RAM int8_t dynLpfFilter = DYN_LPF_NONE;
+static FAST_RAM_ZERO_INIT float dynLpfIdle;
+static FAST_RAM_ZERO_INIT float dynLpfIdlePoint;
+static FAST_RAM_ZERO_INIT float dynLpfInvIdlePoint;
+static FAST_RAM_ZERO_INIT int16_t dynLpfDiff;
+static FAST_RAM_ZERO_INIT uint16_t dynLpfMin;
+
+static void dynLpfFilterInit()
+{
+    if (gyroConfig()->dyn_lpf_gyro_idle > 0 && gyroConfig()->dyn_lpf_gyro_max_hz > 0) {
+        if (gyroConfig()->gyro_lowpass_hz > 0 ) {
+            dynLpfMin = gyroConfig()->gyro_lowpass_hz;
+            switch (gyroConfig()->gyro_lowpass_type) {
+                case FILTER_PT1:
+                    dynLpfFilter = DYN_LPF_PT1;
+                    break;
+                case FILTER_BIQUAD:
+                    dynLpfFilter = DYN_LPF_BIQUAD;
+                    break;
+                default:
+                    dynLpfFilter = DYN_LPF_NONE;
+                    break;
+            }
+        }
+    } else {
+        dynLpfFilter = DYN_LPF_NONE;
+    }
+    dynLpfIdle = gyroConfig()->dyn_lpf_gyro_idle / 100.0f;
+    dynLpfIdlePoint = (dynLpfIdle - (dynLpfIdle * dynLpfIdle * dynLpfIdle) / 3.0f) * 1.5f;
+    dynLpfInvIdlePoint = 1 / (1 - dynLpfIdlePoint);
+    dynLpfDiff = gyroConfig()->dyn_lpf_gyro_max_hz - dynLpfMin;
```

Comment:
as in pid.c, you do not need dynLpfDiff global because its used only in one place, rework dynLpfIdlePoint to dynLpfInvIdlePointScaled and use it later, it wil save some cpu time too
`dynLpfInvIdlePointScaled = 1 / (1 - dynLpfIdlePoint) * (pidProfile->dyn_lpf_gyro_max_hz - dynLpfMin);`